mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -815,16 +862,26 @@ void mav_vehicle::request_mission_item(uint16_t item_id)
     print_verbose("Mission waypoint requested\n");
 }
 
-void mav_vehicle::arm_throttle()
+void mav_vehicle::arm_throttle(bool arm_disarm)
 {
     send_cmd_long(MAV_CMD_COMPONENT_ARM_DISARM, float(arm_disarm), 0, 0, 0, 0,
                   0, 0, request_intervals_ms::arm_disarm);
 }
 
 void mav_vehicle::takeoff()
 {
-    send_cmd_long(MAV_CMD_NAV_TAKEOFF, 0, 0, 0, 0, 0, 0,
-                  defaults::takeoff_init_alt_m, request_intervals_ms::takeoff);
+    switch (this->flight_stack_type) {
+    case firmware_type::PX4:
+        this->set_mode(mode::TAKEOFF);
+        break;
+    case firmware_type::APM:
+        send_cmd_long(MAV_CMD_NAV_TAKEOFF, 0, 0, 0, 0, 0, 0,
+                      defaults::takeoff_init_alt_m,
+                      request_intervals_ms::takeoff);
+        break;
+    default:
+        break;
+    }
```

Comment:
I remember coverity complaining because of mission default cases, but I'm not sure. I'll take a look at it.